### PR TITLE
backwards compatible tag support for graphite, librato, logger and statsd (kinda)

### PR DESCRIPTION
**Pull request by ghost**
### Why
- graphite has tag support now: https://graphite.readthedocs.io/en/latest/tags.html
- librato does too: https://www.librato.com/docs/api/#measurements
- The situation for statsd is a bit murky since the project hasn't an official standard for tags. It seems several 3rd party vendors implemented some solutions. The only tangible solution I found is the Datadog tag format which is also understood by the statsd plugin of https://github.com/influxdata/telegraf. I'd be happy to implement more formats if someone points me to the documentation or source code.
- Unfortunately I haven't found any documentation on tag support in ganglia so I presume ganglia doesn't support tags. Here too I'd implement support if anyone could point me to documentation or source code.

### What
This patch brings tag support to metrology. Each metric can have multiple tags associated with it and metrics with the same name but different tags are distinct from one another.

All public method signatures are backwards compatible. This is achieved by reusing the name parameter in that it can now be a string as previously or a dict containing the name and tags.
I've implemented an alternate version where tags are an optional parameter but didn't like the overall look of this (https://github.com/sv-fko/metrology/tree/tag_support)

This is what it looks like in code:
```
c = self.registry.counter({
   "name": "orders_processed",
   "product": "widgets",
   "payment_method": "credit_card" 
 })
```

Reporter implementations can optin to the tag support by consuming the with_tags property of the registry while old reporters can still just use the iterator on the registry itself which just returns name and value like previously.

### Drive-by changes
- I've removed the profiler method from the Metrology class since it seems to be defunct
- I've extended the sanitizer of the graphite reporter by digging up the rules from graphite

### Open issues
- The librato support is untested since I don't have access to librato.
- Documentation has to reflect the changes

Please let me know what you think. If you would be willing to merge this, I'd update the docs.

### CODE REVIEW

**Comment by cyberdelia** [comment]
That’s a very nice addition, I’ll need some time to review it. If anyone else want to review it they are welcome to as well.

**Comment by sv-fko** [comment]
Thank you for your consideration. I've updated the README to complete the patch for review.

File: metrology/registry.py [hunk 1]
```diff
@@ -53,19 +53,22 @@ def derive(self, name):
 
     def get(self, name):
         with self.lock:
-            return self.metrics[name]
+            key = self._compose_key(name)
+            return self.metrics[key]
 
     def add(self, name, metric):
```
Comment (cyberdelia) [inline metrology/registry.py]: Why not adding an extra parameters for tags?

```suggestion
    def add(self, name, metric, tags = []):
```

Comment (sv-fko) [reply metrology/registry.py]: That's what I did in the other branch I mentioned in the pull request. I didn't like how it's dragging through the whole project and leads to some suboptimal parameter orders for the function signatures. But if you like this style better, I can pull request this other branch :)
Although it needs some additional work.

**Pull request merged**